### PR TITLE
Add NATS DSN

### PR DESCRIPTION
**Pull request by dsodx**
## Change Summary

Added NATS DSN validation.

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: pydantic/networks.py [hunk 1]
```diff
@@ -131,6 +131,8 @@ def __hash__(self) -> int:
 """A type that will accept any MongoDB DSN."""
 KafkaDsn = Annotated[Url, UrlConstraints(allowed_schemes=['kafka'], default_host='localhost', default_port=9092)]
 """A type that will accept any Kafka DSN."""
+NatsDsn = Annotated[MultiHostUrl, UrlConstraints(allowed_schemes=['nats', 'tls', 'ws'], default_host='localhost', default_port=4222)]
+"""A type that will accept any NATS DSN."""
```
Comment (samuelcolvin) [inline pydantic/networks.py:135]: we need to add better documentation for all these types, but while you're adding this, please could you expand the documentation a bit here.

In particular a link to NATS and some context on what it is would be useful.

I for one have no idea what "NATS" is in this context.

**Review by samuelcolvin — CHANGES_REQUESTED** [review]
Thanks so much for this.

also need to add at least one test.

**Comment by dsodx** [comment]
Yes, it's still in progress.

File: pydantic/networks.py (hunk 1 above)
Comment (dsodx) [reply pydantic/networks.py:135]: Ok. I agree it would be better.

**Comment by alexmojaki** [comment]
Looks like this was ready but got lost in a sea of notifications and forgotten. @dsodx can I suggest resolving the merge conflicts?

**Review by sydney-runkle — APPROVED** [review]
Looks great @dsodx,

Thanks for your contribution 😄.

**Comment by dsodx** [comment]
Thank you :smiley: .

**Pull request merged**